controllers/providers/pxc/applier.go

```diff
@@ -95,6 +95,12 @@ func (p *applier) Engine() error {
 		pxc.Spec.PXC.PodSpec.Resources.Limits[corev1.ResourceMemory] = p.DB.Spec.Engine.Resources.Memory
 		pxc.Spec.PXC.PodSpec.Resources.Requests[corev1.ResourceMemory] = p.DB.Spec.Engine.Resources.Memory
 	}
+	// We preserve the settings for existing DBs, otherwise restarts are seen when upgrading Everest.
+	// TODO: Remove this once we figure out how to apply such spec changes without automatic restarts.
+	// See: https://perconadev.atlassian.net/browse/EVEREST-1413
+	if p.DB.Status.Status == everestv1alpha1.AppStateReady {
+		pxc.Spec.PXC.PodSpec.Resources = p.currentPerconaXtraDBClusterSpec.PXC.PodSpec.Resources
```

Comment:
@mayankshah1607 Doesn't this mean that users won't be able to change the resources of a DB that's in the Ready state? With this change we are breaking the feature of vertical scaling, right?

Reply:
@recharte you're right...kinda.

I didn't think of this earlier, that by preserving the settings as-is when the state is "running", a user cannot apply his desired changes anymore.

So I tested it. And turns out, that it still works, unexpectedly. I was able to vertically scale the cluster even though the code here preserves the original settings when the state is running. Reason is that when you vertically scale, it also changes the proxy pod resources. And as a result, the DB goes into the "initialising" state, and the new settings for the pxc pod also get applied (as a side-effect).

Basically, it still "accidentally" works.. This still feels wrong to me, and I'd like it to work based on expectations and not surprises. So I created a proper fix for it: https://github.com/percona/everest-operator/pull/513

Reply:
> it also changes the proxy pod resources
I think this might only true for PXC and not for the other engines